core/types/receipt.go

```diff
@@ -587,7 +587,7 @@ func (rs Receipts) DeriveFields(config *params.ChainConfig, hash common.Hash, nu
 			rs[i].L1GasPrice = gasParams.l1BaseFee
 			rs[i].L1BlobBaseFee = gasParams.l1BlobBaseFee
 			rs[i].L1Fee, rs[i].L1GasUsed = gasParams.costFunc(txs[i].RollupCostData())
-			if txs[i].GasPrice().Cmp(big.NewInt(0)) == 0 && config.IsWright(time) {
+			if txs[i].inner.effectiveGasPrice(new(big.Int), baseFee).Cmp(big.NewInt(0)) == 0 && config.IsWright(time) {
```

Comment:
`rs[i].EffectiveGasPrice = txs[i].inner.effectiveGasPrice(new(big.Int), baseFee)`
line 535 has already calculated this, why not reuse it directly?

Reply:
The gasprice in the bugGas logic is already the effective gasprice, that logic is simply reused when processing the receipt. However, the gasprice in the receipt is the gasprice of tx itself, and is not processed by effecitve gasprice. It needs to be converted through this method.

Reply:
Make sense. Fixed it.